wsapi.go

```diff
@@ -401,12 +395,49 @@ func (s *Session) onVoiceStateUpdate(se *Session, st *VoiceStateUpdate) {
 	// user just ignore it.
 	// TODO Move this IF to the event() func
 	if st.UserID != self.ID {
+		var exists bool
+		var guild *Guild
+
+		for _, guild = range s.State.Guilds {
+			if guild.ID == channel.GuildID {
+				exists = true
+				break
+			}
+		}
+
+		if !exists {
+			return
+		}
+
+		// Handle Leaving Channel
+		if st.ChannelID == "" {
```

Comment:
Can you move this state tracking code into state.go?

Also you can do:

```
	for i, state := range guild.VoiceStates {
		if state.UserID == st.UserID {
			guild.VoiceStates = append(guild.VoiceStates[:i], guild.VoiceStates[i+1:]...)
                         break
		}
	}
```

Instead of constructing a new array.